### ISSUE TITLE
Lint uses of to/from_ne_bytes

### ISSUE THREAD

**Issue by i509VCB** [rust-lang/rust-clippy#10765]
### What it does

Lint uses of `from_ne_bytes` and `to_ne_bytes` for code that needs to be endian aware. It may be desirable to force a developer to consider the endian which data should be stored in, especially when sending data across a network or to another machine.

### Lint Name

endian_bytes

### Category

restriction

### Advantage

- Useful for contexts where endianness needs to be considered (such as serialization of data).

### Drawbacks

Sometimes it may be desirable to use the host endianness. For example a protocol like Wayland uses host endian.

### Example

```rust
// 0xFE4415
let ferris = u32::from_ne_bytes([0x00, 0x15, 0x44, 0xFE]);
```

Could be written as:

```rust
// 0xFE4415
let ferris = u32::from_le_bytes([0x00, 0x15, 0x44, 0xFE]);
```

**Comment by zperk13** [comment]
I'm thinking there should also be a lint for big endian and little endian functions. They'd both be allowed by default, but if for instance, I'm writing a serialization thing and it always uses little endian (never big endian or native endian), it'd be nice to have a lint that stops me from accidentally using big endian or native endian.

**Comment by Centri3** [comment]
> I'm thinking there should also be a lint for big endian and little endian functions. They'd both be allowed by default, but if for instance, I'm writing a serialization thing and it always uses little endian (never big endian or native endian), it'd be nice to have a lint that stops me from accidentally using big endian or native endian.

I was looking into this lint before and wanted to add exactly this. I got `to_xx_bytes` working but not `from_xx_bytes`, but seemed to forgot to claim it :woman_shrugging:

@rustbot claim
